build-monitor-plugin/src/main/java/com/smartcodeltd/jenkinsci/plugins/buildmonitor/viewmodel/BuildView.java

```diff
@@ -107,6 +115,21 @@ public String description() {
         return getOrElse(build.getDescription(), "");
     }
 
+    @Override
+    public List<String> stages() {
+        if (this.build instanceof WorkflowRun) {
+            WorkflowRun currentBuild = (WorkflowRun) this.build;
+
+            if (currentBuild.isBuilding()) {
+                StageTraversal traversal = new StageTraversal();
+                traversal.start(currentBuild.getExecution().getCurrentHeads());
```

Comment:
Jenkins v2.45 with Pipeline v2.5.

I'm not sure which of our Pipeline jobs that caused the issue so I'm unsure on what info to give you. However, I no longer believe it is when the job is not running (that should be covered by your .isBuilding() check). Instead I suspect it's a corner-case where the job is between phases or something. Perhaps as it just finishing up or something, where technically the job is still building per currentBuild.isBuilding(), but the execution is no longer active.

We have a lot of plugins that do stuff like calculate workspace usage, etc, that's triggered after builds. I have no idea how they work, but potentially they can cause currentBuild.isBuilding() to return true, while currentBuild.getExecution() is null. Or something :)